server/api/settings.py

```diff
@@ -60,19 +62,28 @@
 ]
 
 MIDDLEWARE = [
+    "corsheaders.middleware.CorsMiddleware",
+    "django.middleware.common.CommonMiddleware",
     "django.middleware.security.SecurityMiddleware",
     "django.contrib.sessions.middleware.SessionMiddleware",
-    "django.middleware.common.CommonMiddleware",
     "django.middleware.csrf.CsrfViewMiddleware",
     "django.contrib.auth.middleware.AuthenticationMiddleware",
     "django.contrib.messages.middleware.MessageMiddleware",
     "django.middleware.clickjacking.XFrameOptionsMiddleware",
-    "corsheaders.middleware.CorsMiddleware",
 ]
 
+CORS_ORIGIN_WHITELIST = (
+    'http://localhost:3000'
+    "http://127.0.0.1:3000",
+    "http://127.0.0.1:3001",
+    "http://localhost:3001",
+)
+
 CORS_ALLOWED_ORIGINS = [
```

Comment:
the better practice is to use the cors-heasers to handle the CORS.
Please check this link out: [Handling CORS in Django Rest Framework: A Comprehensive Guide](https://medium.com/django-unleashed/handling-cors-in-django-rest-framework-a-comprehensive-guide-ec11b0bc6807)

Reply:
will remove this